webview/platforms/edgechromium.py

```diff
@@ -0,0 +1,204 @@
+from System.Drawing import Size, Point, Icon, Color, ColorTranslator, SizeF
+
+archpath = 'x64' if architecture()[0] == '64bit' else 'x86'
+os.environ['Path'] = interop_dll_path(archpath) + ';' + os.environ['Path']
+clr.AddReference(interop_dll_path('Microsoft.Web.WebView2.Core.dll'))
+clr.AddReference(interop_dll_path('Microsoft.Web.WebView2.WinForms.dll'))
+from Microsoft.Web.WebView2.WinForms import WebView2, CoreWebView2CreationProperties
+from Microsoft.Web.WebView2.Core import CoreWebView2Environment
+
+logger = logging.getLogger('pywebview')
+
+class EdgeChrome:
+    def __init__(self, form, window):
+        self.pywebview_window = window
+        self.web_view = WebView2()
+        props = CoreWebView2CreationProperties()
+        #props.UserDataFolder = os.path.join(os.getcwd(), 'profile')
+        props.UserDataFolder = os.path.join(os.environ['LOCALAPPDATA'], 'pywebview')
+        self.web_view.CreationProperties = props
+        form.Controls.Add(self.web_view)
+
+        self.js_result_semaphore = Semaphore(0)
+        self.web_view.Dock = WinForms.DockStyle.Fill
+        #settings under on_webview_ready 
+        self.web_view.CoreWebView2Ready += self.on_webview_ready
+        self.web_view.NavigationStarting += self.on_navigation_start
+        self.web_view.NavigationCompleted += self.on_navigation_completed
+        self.web_view.WebMessageReceived += self.on_script_notify
+
+        self.httpd = None # HTTP server for load_html
+        self.tmpdir = None
+        self.url = None
+        self.ishtml = False
+
+        if window.html or 'localhost' in window.real_url or '127.0.0.1' in window.real_url:
+            _allow_localhost()
+
+        if window.real_url:
+            self.load_url(window.real_url)
+        elif window.html:
+            self.load_html(window.html, '')
+        else:
+            self.load_html(default_html, '')
+
+    def evaluate_js(self, script, callback=None):
+        def _callback(result):
+            if callback is None:
+                self.js_result = None if result is None or result == '' else json.loads(result)
+                self.js_result_semaphore.release()
+            else: 
+                # future js callback option to handle async js method
+                callback(result)
+                self.js_result = None
+                self.js_result_semaphore.release()
+
+        self.syncContextTaskScheduler = TaskScheduler.FromCurrentSynchronizationContext()
+        try:
+            result = self.web_view.ExecuteScriptAsync(script).ContinueWith(
+            Action[Task[String]](
+                lambda task: _callback(json.loads(task.Result))
+            ),
+            self.syncContextTaskScheduler)
+        except Exception as e:
+            logger.exception('Error occurred in script')
+            self.js_result = None
+            self.js_result_semaphore.release()
+
+    def get_current_url(self):
+        return self.url
+
+    def load_html(self, html, base_uri):
```

Comment:
HTML is used for loading HTML in EdgeHTML, because of the bug in EdgeHTML which crashes regular html loading. I guess this is not a case in Chromium, so regular html loading can be used instead.

Reply:
yes